python/comps.py

```diff
@@ -257,7 +256,7 @@ def _get_top_n_comps(
             similarity_score = 0.0
             for tree_idx in range(len(leaf_node_matrix[x_i])):
                 similarity_score += (
-                    weights[tree_idx] * (
+                    weights_matrix[x_i][tree_idx] * (
```

Comment:
Based on my local testing, this is slightly slower than reusing the same weights vector for every observation, but only by a small amount. We'll see how the runtime on Batch compares.